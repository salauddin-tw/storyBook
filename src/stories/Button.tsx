import React from "react"
import styled from "styled-components"
import "./button.css"

interface ButtonProps {
  /**
   * Is this the principal call to action on the page?
   */
  primary?: boolean
  /**
   * What background color to use
   */
  backgroundColor?: string
  /**
   * How large should the button be?
   */
  size?: "small" | "medium" | "large"
  /**
   * Button contents
   */
  label: string
  /**
   * Optional click handler
   */
  onClick?: () => void
}

const StyledButton = styled.button`
  font-family: "Nunito Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
  font-weight: 800;
  border: 0;
  border-radius: 3em;
  cursor: pointer;
  display: inline-block;
  line-height: 1;
  &:hover {
    background-color: #0056b3;
  }
`

/**
 * Primary UI component for user interaction
 */
export const Button = ({
  primary = false,
  size = "medium",
  backgroundColor,
  label,
  ...props
}: ButtonProps) => {
  const mode = primary
    ? "storybook-button--primary"
    : "storybook-button--secondary"
  return (
    <StyledButton
      className={[`storybook-button--${size}`, mode].join(" ")}
      style={{ backgroundColor }}
      {...props}
    >
      {label}
    </StyledButton>
  )
}
