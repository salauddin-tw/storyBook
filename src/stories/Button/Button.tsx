/* eslint-disable */
import React from "react"
import styled, { css } from "styled-components"

interface ButtonProps {
  /**
   * Is this the principal call to action on the page?
   */
  primary?: boolean
  /**
   * What background color to use
   */
  backgroundColor?: string
  /**
   * How large should the button be?
   */
  size?: "small" | "medium" | "large"
  /**
   * Button contents
   */
  label: string
  /**
   * Optional click handler
   */
  onClick?: () => void
}

const StyledButton = styled.button<{
  $primary?: boolean
  buttonSize: "small" | "medium" | "large"
}>`
  font-family: "Nunito Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
  font-weight: 800;
  border-radius: 12px;

  cursor: pointer;
  background-color: ${(props) => (props.$primary ? "#ED1C24" : "white")};
  color: ${(props) => (props.$primary ? "white" : "#ED1C24")};
  border-color: ${(props) => (props.$primary ? "#ED1C24" : "grey")};
  border: ${(props) => (props.$primary ? 0 : "1px solid")};
  ${({ buttonSize }) => {
    if (buttonSize === "small")
      return css`
        font-size: 12px;
        padding: 10px 16px;
      `
    else if (buttonSize === "large")
      return css`
        font-size: 16px;
        padding: 12px 32px;
      `
    else
      return css`
        font-size: 14px;
        padding: 12px 24px;
      `
  }}// &:hover {
  //   background-color: #0056b3;
  // }
`

/**
 * Primary UI component for user interaction
 */
const Button = ({
  primary = false,
  size = "medium",
  backgroundColor,
  label,
  ...props
}: ButtonProps) => {
  return (
    <StyledButton
      $primary={primary}
      buttonSize={size}
      style={{ backgroundColor }}
      {...props}
    >
      {label}
    </StyledButton>
  )
}

export default Button;