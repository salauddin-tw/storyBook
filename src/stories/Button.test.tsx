import React from "react"
import { render } from "@testing-library/react"
import { Button } from "./Button"

describe("Button test", () => {
  it("should render button", () => {
    const { container } = render(<Button label="test" />)
    expect(container).toMatchSnapshot()
  })
})
