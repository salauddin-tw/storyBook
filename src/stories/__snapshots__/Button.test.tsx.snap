// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test should render button 1`] = `
<div>
  <button
    class="sc-aYaIB eCpUyc storybook-button--medium storybook-button--secondary"
  >
    test
  </button>
</div>
`;
